llvm/test/CodeGen/PowerPC/aix-tls-gd-double.ll

```diff
@@ -156,10 +156,11 @@ define void @storesTIInit(double %Val) #0 {
 ; SMALL32:       # %bb.0: # %entry
 ; SMALL32-NEXT:    mflr 0
 ; SMALL32-NEXT:    stwu 1, -32(1)
-; SMALL32-NEXT:    lwz 3, L..C4(2) # target-flags(ppc-lo) @TIInit
-; SMALL32-NEXT:    lwz 4, L..C5(2) # target-flags(ppc-tlsgd) @TIInit
 ; SMALL32-NEXT:    stw 0, 40(1)
-; SMALL32-NEXT:    bla .__tls_get_addr[PR]
+; SMALL32-NEXT:    lwz 6, L..C4(2) # target-flags(ppc-tlsld) @TIInit
```

Comment:
Can we use `r4` here? That should align with the assembler manual and save register use.

Reply:
Updated a version to address the issue, will continue improve this point.

Reply:
New approach will invalidate [d400de0](https://github.com/llvm/llvm-project/pull/66316/commits/d400de0cd3ffbb80794961a3443a191760b32aa1), please ignore that.

Reply:
Now for the small-model, we did reuse r4. However for the large-model, since machine-scheduler will decide to move the ADDIS before the .__tls_get_mod call, I did not try to move ADDIS in PPCTLSDynamicCall.cpp